### PR TITLE
CLOUDP-120668: Add register command skeleton

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ [CLOUDP-120668](https://jira.mongodb.org/browse/CLOUDP-120668)

```shell
➜  mongocli git:(CLOUDP-120668) atlasdev auth register --help
Register with MongoDB Atlas.

Usage:
  atlas auth register [flags]

Examples:
  To start the interactive setup:
  $ atlas auth register


Flags:
      --gov          Register to Atlas for Government.
  -h, --help         help for register
      --noBrowser    Don't try to open a browser session.
      --skipConfig   Skip profile configuration.

Global Flags:
  -P, --profile string   Profile to use from your configuration file.
```

```shell
➜  mongocli git:(CLOUDP-120668) atlasdev auth register
Create and verify your MongoDB Atlas account from the web browser and return to Atlas CLI after activation.
```

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/auth/register.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	// TODO: CLOUDP-120669
+	return nil
+}
+
+func RegisterBuilder() *cobra.Command {
+	opts := &registerOpts{}
+	cmd := &cobra.Command{
+		Use:    "register",
+		Short:  "Register with MongoDB Atlas.",
+		Hidden: true,
```
Comment (blva) [inline internal/cli/auth/register.go:40]: the command is hidden, no doc generated until we flip it.

File: internal/cli/auth/login_test.go [hunk 2]
```diff
@@ -22,6 +22,8 @@ import (
 	"context"
 	"testing"
 
+	"github.com/mongodb/mongocli/internal/config"
+
```
Comment (gssbzn) [inline internal/cli/auth/login_test.go:26]:
```suggestion
	"github.com/mongodb/mongocli/internal/config"
```

**Review by gssbzn — COMMENTED** [review]
some small comments but in general LGTM

File: internal/cli/auth/login_test.go [hunk 3]
```diff
@@ -40,6 +42,22 @@ func TestBuilder(t *testing.T) {
 	)
 }
 
+func TestBuilderForAtlas(t *testing.T) {
+	// Test for AtlasCLI
+	prevTool := config.ToolName
+	config.ToolName = config.AtlasCLI
+	defer func() {
+		config.ToolName = prevTool
+	}()
```
[the remark this answers is not in the document]
Comment (blva) [reply internal/cli/auth/login_test.go]: Thanks! updated

**Review by gssbzn — DISMISSED** [review]
LGTM

File: internal/cli/auth/register.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+		Example: fmt.Sprintf(`  To start the interactive setup:
+  $ %s auth register
+`, config.BinName()),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if hasUserProgrammaticKeys() {
+				return fmt.Errorf(`you have already set the programmatic keys for this profile. 
+
+Run '%s auth register --profile <profileName>' to use your username and password on a new profile`, config.BinName())
```
Comment (JuliaMongo) [inline internal/cli/auth/register.go:48]: with a new profile

File: internal/cli/auth/register_test.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+	opts := &registerOpts{
+		*loginOpts,
+	}
+
+	opts.OutWriter = buf
+	opts.Store = mockStore
+
+	require.NoError(t, opts.Run())
+	assert.Equal(t, `Create and verify your MongoDB Atlas account from the web browser and return to Atlas CLI after activation.
```
Comment (JuliaMongo) [inline internal/cli/auth/register_test.go:63]: after activating your account. (instead of "after activation").

**Review by JuliaMongo — DISMISSED** [review]
LGTM, Bianca! Left two small comments.

**Review by colm-quinn — APPROVED** [review]
LGTM!

**Review by JuliaMongo — COMMENTED** [review]
LGTM

**Pull request merged**